node-graph/gcore/src/raster.rs

```diff
@@ -53,6 +53,32 @@ impl<N: Node<(), Output = f32> + Copy> BrightenColorNode<N> {
 	}
 }
 
+#[derive(Debug, Clone, Copy)]
+pub struct GammaColorNode<N: Node<(), Output = f32>>(N);
+
+impl<N: Node<(), Output = f32>> Node<Color> for GammaColorNode<N> {
+	type Output = Color;
+	fn eval(self, color: Color) -> Color {
+		let gamma = self.0.eval(());
+		let per_channel = |col: f32| col.powf(gamma);
+		Color::from_rgbaf32_unchecked(per_channel(color.r()), per_channel(color.g()), per_channel(color.b()), color.a())
+	}
+}
+impl<N: Node<(), Output = f32> + Copy> Node<Color> for &GammaColorNode<N> {
+	type Output = Color;
+	fn eval(self, color: Color) -> Color {
+		let gamma = self.0.eval(());
+		let per_channel = |col: f32| col.powf(gamma);
+		Color::from_rgbaf32_unchecked(per_channel(color.r()), per_channel(color.g()), per_channel(color.b()), color.a())
```

Comment:
Could we use the `node_fn` macro to avoid this code duplication?

Reply:
Yes, will fix